pyproject.toml

```diff
@@ -1,22 +1,82 @@
-[tool.black]
-target-version = ['py37', 'py38', 'py39']
+[project]
+name = "geographer"
+version = "0.1"
+description = "Build object-centric remote sensing computer vision datasets"
+readme = "README.md"
+license = {file = "LICENSE"}
+authors = [
+    {name = "Rustam Antia", email = "rustam.antia@gmail.com"}
+]
+keywords = ["remote sensing", "computer vision", "satellite imagery", "GIS", "machine learning"]
+classifiers = [
+    "License :: Other/Proprietary License",
+    "Programming Language :: Python :: 3"
+]
+requires-python = ">=3.9"
+dependencies = [
+    "eodag",
+    "eval_type_backport",
+    "fiona",
+    "geojson",
+    "geopandas",
+    "GitPython",
+    "ipywidgets",
+    "matplotlib",
+    "networkx",
+    "numpy",
+    "packaging",
+    "pandas",
+    "pydantic >= 2.0",
+    "pyproj",
+    "rasterio",
+    "requests",
+    "rioxarray",
+    "rtree",
+    "scipy",
+    "sentinelsat",
+    "Shapely",
+    "tqdm",
+    "urllib3"
+]
 
-[tool.isort]
-atomic = true
-profile = "black"
-skip_gitignore = true
-
-[tool.mypy]
-plugins = ["pydantic.mypy"]
-show_error_codes = true
-disallow_untyped_defs = false
-disallow_incomplete_defs = false
-ignore_missing_imports = true
-warn_unused_ignores = false
-warn_return_any = false
-warn_unreachable = false
-strict_optional=true
+[project.optional-dependencies]
+dev = [
+    "build",
+    "docformatter",
+    "ipykernel",
+    "pytest",
+    "ruff"
```

Comment:
I recommend pinning the exact ruff version, since the formatting or linting can change slightly between versions leading to funny diffs.